### PR TITLE
audacity: update to 3.3.3.

### PR DESCRIPTION
**Pull request by ii8**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures:
  - armv6l-musl
  - i686
  - armv7l

### CODE REVIEW

File: srcpkgs/audacity/template [hunk 1]
```diff
@@ -1,39 +1,38 @@
...
-if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
-	LDFLAGS="-latomic"
-fi
-
-CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+pre_build() {
+	if [ "$CROSS_BUILD" ]; then
+		mkdir -p ${wrksrc}/${build_wrksrc}/scripts
+		echo "#!/bin/sh" > ${wrksrc}/${build_wrksrc}/scripts/image-compiler
+		echo "/usr/bin/qemu-${XBPS_TARGET_QEMU_MACHINE}-static ${wrksrc}/${build_wrksrc}/Release/bin/image-compiler \$@" >> ${wrksrc}/${build_wrksrc}/scripts/image-compiler
```
Comment (ahesford) [inline srcpkgs/audacity/template:30]: This should be an indented HEREDOC for legibility.
```suggestion
		cat > "${XBPS_WRAPPERDIR}/image-compiler" <<-EOF
			#!/bin/sh
			/usr/bin/qemu-${XBPS_TARGET_QEMU_MACHINE}-static "${wrksrc}/${build_wrksrc}/Release/bin/image-compiler" "\$@"
			EOF
```

File: srcpkgs/audacity/template [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
+		chmod +x ${wrksrc}/${build_wrksrc}/scripts/image-compiler
+		export PATH="${wrksrc}/${build_wrksrc}/scripts/:$PATH"
```
Comment (ahesford) [inline srcpkgs/audacity/template:32]: No need for this anymore.

File: srcpkgs/audacity/template [hunk 3]
```diff
@@ -1,39 +1,38 @@
...
-	LDFLAGS="-latomic"
-fi
-
-CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
+pre_build() {
+	if [ "$CROSS_BUILD" ]; then
+		mkdir -p ${wrksrc}/${build_wrksrc}/scripts
+		echo "#!/bin/sh" > ${wrksrc}/${build_wrksrc}/scripts/image-compiler
+		echo "/usr/bin/qemu-${XBPS_TARGET_QEMU_MACHINE}-static ${wrksrc}/${build_wrksrc}/Release/bin/image-compiler \$@" >> ${wrksrc}/${build_wrksrc}/scripts/image-compiler
+		chmod +x ${wrksrc}/${build_wrksrc}/scripts/image-compiler
```
Comment (ahesford) [inline srcpkgs/audacity/template:31]: Use full numeric permissions for predictability.
```suggestion
		chmod 0755 "${XBPS_WRAPPERDIR}/image-compiler"
```

**Review by ahesford — CHANGES_REQUESTED** [review]

File: srcpkgs/audacity/template [hunk 4]
```diff
@@ -1,39 +1,38 @@
...
-case "$XBPS_TARGET_MACHINE" in
-	i686*|x86_64*) ;;
-	*) configure_args+=" --disable-sse" ;;
-esac
-
-if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
-	LDFLAGS="-latomic"
-fi
-
-CXXFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64"
```
[the remark this answers is not in the document]
Comment (ii8) [reply srcpkgs/audacity/template:31]: `_LARGEFILE_SOURCE` is redundant. I've added `_FILE_OFFSET_BITS=64` again only for those targets where it has an effect.

**Comment by ii8** [comment]
The i686 build is failing, it uses ninja which depends on python and the python tests are failing.
Should I just use make instead of ninja:
```sh
if [ "$XBPS_TARGET_MACHINE" = i686 ]; then                                      
        export CMAKE_GENERATOR="Unix Makefiles"                                 
        make_cmd=make                                                           
fi
```
or is there a better way to fix that?

**Comment by ahesford** [comment]
Your test is failing because we are in the middle of a massive Python update; the problem is ephemeral and not related to this PR.

File: srcpkgs/audacity/template (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahesford) [reply srcpkgs/audacity/template:31]: There's no need for the guard; we have several templates that add these flags, and we always just add them to CXXFLAGS globally.

**Comment by ii8** [comment]
Previous attempts for reference: #44662,  #40508

**Pull request merged**